tree_segment.py

```diff
@@ -0,0 +1,434 @@
+import laspy
+import matplotlib.pyplot as plt
+import math
+import numpy as np
+import random
+import time
+import pandas as pd
+import argparse
+import re
+import os
+
+
+class Point_Cloud:
+    # object that stores the points in a point cloud, initialized from a las file
+    # points can be enabled or disabled, disabled points are ignored when searching for stems
+    def __init__(self, file_name, verbose=False):
+        all_data = laspy.read(file_name)
+        height_of_points_above_ground = np.array(
+            [point[-1] for point in all_data.points.array])
+        self.points = [Point(xyz, height) for xyz, height in zip(
+            all_data.xyz, height_of_points_above_ground)]
+        # internal, to access this use enabled_points()
+        self.__enabled_points__ = self.points
```

Comment:
Use "_variable_name" for private variables: https://google.github.io/styleguide/pyguide.html#381-docstrings

In particular, "__double_leading_and_trailing_underscore__" names are reserved by Python.

Reply:
Ack! I knew some amount of underscores was appropriate, I guess I just got overenthusiastic.